### PR TITLE
Back out "[PyPer][BE] Fix test_scripted_module in StatCollector"

### PR DESCRIPTION
**Pull request by PaulZhang12**
Differential Revision: D48908507

### CODE REVIEW

**Review by jerryzh168 — APPROVED** [review]

File: torch/ao/quantization/observer.py [hunk 1]
```diff
@@ -1183,6 +1183,9 @@ def forward(self, x_orig: torch.Tensor) -> torch.Tensor:
             assert (
                 combined_min.numel() == 1 and combined_max.numel() == 1
             ), "histogram min/max values must be scalar."
+
+            combined_min, combined_max = combined_min.detach(), combined_max.detach()
```
Comment (jerryzh168) [inline torch/ao/quantization/observer.py:1187]: comment?

Comment (PaulZhang12) [reply torch/ao/quantization/observer.py:1187]: TorchScript causes combined_min, combined_max to requires_grad even though it shouldn't based on upstream tensors.

File: torch/ao/quantization/observer.py [hunk 2]
```diff
@@ -1183,6 +1183,11 @@ def forward(self, x_orig: torch.Tensor) -> torch.Tensor:
             assert (
                 combined_min.numel() == 1 and combined_max.numel() == 1
             ), "histogram min/max values must be scalar."
+
+            # TODO: For some reason, this is required for it to pass torchscript test
```
Comment (jerryzh168) [inline torch/ao/quantization/observer.py:1187]: thanks, the current code didn't do detach() actually, maybe you can add the detach() back as well
